### ISSUE TITLE
[Discussion] About new components Label and LabelList

### ISSUE THREAD

**Issue by xile611** [recharts/recharts#522]
Recharts has many issues about labels now:

#160 Simpler custom labels
#191 Label in center of PieChart
#472  "Total" bar label for stacked bar chart
#466  Rotated Axis labels

and the labels of components diffs very much now. Meanwhile, some simple features like rotate, location,  can only be implemented by customized labels.

So I try to refactor labels and create two new components: `Label`, `LabelList` (create an array of labels), then we can create labels like the following:

```jsx
  <Line>
    <LabelList position="bottom" /> 
  </Line>

  <XAxis>
    <Label angle={45} position="right">This is a label</Label>
  </XAxis>
```

Then the props `label` of `Line\Area|Bar...` will be removed.
Let me know if you have some different thoughts.

[Code of Label](https://github.com/recharts/recharts/blob/feat/label/src/component/Label.js)
[Code of LabelList](https://github.com/recharts/recharts/blob/feat/label/src/component/LabelList.js)

**Comment by super-cache-money** [comment]
Can't wait for this...

**Comment by Eric24** [comment]
It looks interesting, but it's hard to tell how it would be used without an example--is that possible?

**Comment by xile611** [comment]
@Eric24 You can check out the branch `feat/label` to have  a test.

**Comment by Eric24** [comment]
Yes, I think this is a good step, but I would not deprecate the label props in Line/Area/Bar... I see these new components as being a more advanced option, but sometimes just the basic label prop is all you need (plus, it would be a breaking change).

**Comment by aconran** [comment]
Looking forward to see these componentized

**Comment by xile611** [comment]
This feature has been released in 1.0.0-alpha.0

**Comment by kamarajuPrathi** [comment]
@xile611 can you provide example for stacked bar chart where do we mention a common label
